tests/suites/test_suite_entropy.function

```diff
@@ -567,20 +572,41 @@ exit:
 }
 /* END_CASE */
 
-/* BEGIN_CASE depends_on:MBEDTLS_PLATFORM_GET_ENTROPY_ALT */
+/* BEGIN_CASE depends_on:MBEDTLS_PLATFORM_C:MBEDTLS_PLATFORM_GET_ENTROPY_ALT */
 void entropy_test_platform_get_entropy_alt(void)
 {
+    mbedtls_entropy_context ctx;
     uint8_t data[MBEDTLS_ENTROPY_POLL_PLATFORM_MIN];
-    size_t olen;
+    size_t olen, call_count;
 
+    /* Simple callback test -> all fine. */
     TEST_ASSERT(mbedtls_entropy_poll_platform(NULL, data, sizeof(data), &olen) == 0);
     TEST_LE_U(1, olen);
     TEST_LE_U(olen, sizeof(data));
-    mbedtls_test_get_entropy_force(1, SIZE_MAX);
+
+    mbedtls_test_platform_get_entropy_reset();
+
+    /* Force a failure -> return MBEDTLS_ERR_ENTROPY_SOURCE_FAILED. */
+    mbedtls_test_platform_get_entropy_set_force_failure(1);
     TEST_ASSERT(mbedtls_entropy_poll_platform(NULL, data, sizeof(data), &olen) ==
                 MBEDTLS_ERR_ENTROPY_SOURCE_FAILED);
-    mbedtls_test_get_entropy_force(0, 8 * (sizeof(data) - 1));
+
+    mbedtls_test_platform_get_entropy_reset();
+
+    /* Return less entropy than expected -> entropy gathering fails */
+    mbedtls_test_platform_get_entropy_set_entropy_content(8 * (sizeof(data) - 1));
     TEST_ASSERT(mbedtls_entropy_poll_platform(NULL, data, sizeof(data), &olen) ==
                 MBEDTLS_ERR_ENTROPY_SOURCE_FAILED);
+
+    mbedtls_test_platform_get_entropy_reset();
+
+    /* Return less data than required -> callback will be called more than once. */
+    mbedtls_entropy_init(&ctx);
+    mbedtls_test_platform_get_entropy_set_output_len(
+        TEST_PLATFORM_GET_ENTROPY_DATA_LENGTH_PER_CALL);
+    TEST_ASSERT(mbedtls_entropy_func(&ctx, data, sizeof(data)) == 0);
+    mbedtls_entropy_free(&ctx);
+    call_count = mbedtls_test_platform_get_entropy_get_call_count();
+    TEST_EQUAL(call_count, TEST_PLATFORM_GET_ENTROPY_EXP_CALL_COUNT);
 }
```

Comment:
Sorry I missed this earlier: if a test assertion fails, it might leave the entropy callback test in a non-default state, which will likely have weird effects on subsequent tests. Please reset the state in the `exit:` block, that's how we normally deal with test state.